src/main/java/ca/corefacility/bioinformatics/irida/ria/web/clients/ClientsController.java

```diff
@@ -271,11 +285,11 @@ public String postEditClient(@PathVariable Long clientId,
 
 		String response;
 		try {
-			if (readClient.getAuthorizedGrantTypes()
-					.contains("authorization_code") && registeredRedirectUri.isEmpty()) {
-				//noinspection unchecked
+			if (readClient.getAuthorizedGrantTypes().contains("authorization_code")
+					&& registeredRedirectUri.isEmpty()) {
+				// noinspection unchecked
 				throw new ConstraintViolationException("Redirect URI required for authorization_code clients",
-						Sets.newHashSet(ConstraintViolationImpl.forParameterValidation("client.redirect.required",
+						Collections.singleton(ConstraintViolationImpl.forParameterValidation("client.redirect.required",
```

Comment:
Changed to `Collections.singleton` for instantiating a single object set which removed type warnings.